Updated scheduler docs (tracing) (#37484)
quarkusio/quarkus: docs/src/main/asciidoc/scheduler-reference.adoc

== OpenTelemetry Tracing

> If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present, every job execution, whether using the `@Scheduled` annotation or programmatic scheduling, automatically creates a span named after the job's identity.

Reviewer: ```suggestion
If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then every job execution, either defined with the `@Scheduled` annotation or scheduled programmatically, automatically creates a span named after the job's <<identity>>.
```

`[[identity]]` needs to be added before `=== Identity`...

Reviewer: I think that it's better to add an explicit section id if it's used in a link, e.g. something like `[#identity]`...

== OpenTelemetry Tracing

> If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then every job execution, either defined with the `@Scheduled` annotation or scheduled programmatically, automatically creates a span named after the job's <<identity>>.[[virtual_threads]]

Reviewer: The `[[virtual_threads]]` section id should be on a new line...
